core/src/main/scala/org/apache/spark/ExecutorAllocationManager.scala

```diff
@@ -804,6 +824,14 @@ private[spark] class ExecutorAllocationManager(
       }
     }
 
+    override def onUnpersistRDD(unpersistRDD: SparkListenerUnpersistRDD): Unit = {
```

Comment:
Rather than trigger this extra check for idle executors and plumb through additional fields, how about just checking cached blocks when onExecutorIdle is already called? or is the point that it will never be called if its cache timeout is set to a very long time?

Reply:
onExecutorIdle is called out only when a task ends on a given executor. So unless there is a task which is scheduled on an executor which had previously cached data but currently doesn't, spark will hold on to that executor forever.